src/runtime/fmisc.r

```diff
@@ -595,7 +595,7 @@ function{1} ishift(i,j)
          runerr(101,j);
       cj = IntVal(j);
       if (Type(i) == T_Lrgint || cj >= WordBits
-      || ((ci=(uword)IntVal(i))!=0 && cj>0 && (ci >= (1<<(WordBits-cj-1))))) {
+      || ((ci=(uword)IntVal(i))!=0 && cj>0 && (ci >= (1L<<(WordBits-cj-1))))) {
```

Comment:
Given we are comparing (1L <<( ...)) with an unsigned variable, I think 1uL would be more appropriate rather than relying on the compiler to correctly implement the promotion and integral conversion rules.  Either works (probably), but 1uL makes it marginally clearer.